metricbeat/helper/metricset.go

```diff
@@ -29,14 +31,70 @@ func NewMetricSet(name string, new func() MetricSeter, module *Module) (*MetricS
 }
 
 func (m *MetricSet) Setup() error {
-	// TODO: Call fetch for each host if hosts are set.
+
+	// In case no hosts are set, set at least an empty string
+	// This ensure that also for metricsets where host is not used
+	// That fetch is at least called once
+	if len(m.Config.Hosts) == 0 {
+		m.Config.Hosts = append(m.Config.Hosts, "")
+	}
+
 	// Host is a first class citizen and does not have to be handled by the metricset itself
 	return m.MetricSeter.Setup(m)
 }
 
 // RunMetric runs the given metricSet and returns the event
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
-	// TODO: Call fetch for each host if hosts are set.
-	// Host is a first class citizen and does not have to be handled by the metricset itself
-	return m.MetricSeter.Fetch(m)
+func (m *MetricSet) Fetch() error {
+
+	for _, host := range m.Config.Hosts {
+		// TODO Improve fetching in go routing -> how are go routines stopped if they take too long? - @ruflin,20160314
+		go func(h string) {
+			m.Module.wg.Add(1)
```

Comment:
> I think that the usage of the same WaitGroup in `Module.FetchMetricSets` is unnecessary.

With regard to the above comment, what I meant is that the calls to the module's WaitGroup can simply be removed from `Module.FetchMetricSets` since they are now done in the `Fetch()` method for each metric set.

To be clear, I wasn't suggesting a new waitgroup for each metricset.